### PR TITLE
Change Exception and UI class

### PR DESCRIPTION
**Pull request by Joanna-YJA**
Added and changed a few exception classes. But I couldn't find much to change as the exception classes and ui interface are already there, do suggest if you have any ideas, thank you!
@Abhiman2211

### CODE REVIEW

File: src/main/java/seedu/address/AlfredException.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package seedu.address;
+
+public abstract class AlfredException extends Exception {
+
+
+    public AlfredException(String message) {
+        super(message);
+    }
+
+    public AlfredException(String message, Throwable cause) {
```
Comment (hcwong) [inline src/main/java/seedu/address/AlfredException.java]: Just curious, why have cause?

Comment (Joanna-YJA) [reply src/main/java/seedu/address/AlfredException.java]: It’s an abstract class with all the different constructors for other exception to extend from. Other exceptions need a throwable cause to indicate source of error

File: src/main/java/seedu/address/AlfredException.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        super(message, cause);
+    }
+
+}
```
Comment (hcwong) [inline src/main/java/seedu/address/AlfredException.java]: Is there an extra closing brace here?

File: src/main/java/seedu/address/commons/exceptions/DataConversionException.java [hunk 3]
```diff
@@ -1,9 +1,16 @@
 package seedu.address.commons.exceptions;
 
+import seedu.address.AlfredException;
+
 /**
  * Represents an error during conversion of data from one format to another
  */
-public class DataConversionException extends Exception {
+public class DataConversionException extends AlfredException {
+    /** Constructs an instance of DataConversionException.
```
Comment (hcwong) [inline src/main/java/seedu/address/commons/exceptions/DataConversionException.java]: IIRC this first comment should start on the next line.

File: src/main/java/seedu/address/logic/parser/exceptions/ParseException.java [hunk 4]
```diff
@@ -4,6 +4,12 @@
 
 /**
  * Represents a parse error encountered by a parser.
+ * ParseException is thrown is one of thr 4 scenario:
```
Comment (hcwong) [inline src/main/java/seedu/address/logic/parser/exceptions/ParseException.java]: the*

File: src/main/java/seedu/address/model/person/exceptions/DuplicateEntityException.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package seedu.address.model.person.exceptions;
+
+import seedu.address.AlfredRuntimeException;
+
+/**
+ * Signals that the operation will result in duplicate Entity (Entities are considered duplicates if they have the same
+ * identity).
```
Comment (hcwong) [inline src/main/java/seedu/address/model/person/exceptions/DuplicateEntityException.java]: IMO a bit weird should keep the `identity)` on the same line 6.

File: src/main/java/seedu/address/storage/exceptions/LoadException.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+/**
+ * Represents an error in loading from JSON file or Storage file into AddressBook when the app is reopened.
+ */
+public class LoadException extends AlfredException {
+    public LoadException(String message, Throwable cause) {
+        super(message, cause);
+    }
+
+}
```
Comment (hcwong) [inline src/main/java/seedu/address/storage/exceptions/LoadException.java]: Is there an extra closing brace here?

File: src/main/java/seedu/address/commons/core/Messages.java [hunk 7]
```diff
@@ -6,8 +6,12 @@
 public class Messages {
 
     public static final String MESSAGE_UNKNOWN_COMMAND = "Unknown command";
+    public static final String MESSAGE_UNKNOWN_TYPE = "Unknown entity type";
+    public static final String MESSAGE_INCOMPLETE_INPUT = "User input is incomplete";
     public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n%1$s";
-    public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid";
+    public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The entity index provided is invalid";
     public static final String MESSAGE_PERSONS_LISTED_OVERVIEW = "%1$d persons listed!";
+    //to set restrictions as some operations can only be applied to some entity
```
Comment (hcwong) [inline src/main/java/seedu/address/commons/core/Messages.java]: I think the rest of us have a space between the // and the start of the comment

File: src/main/java/seedu/address/storage/exceptions/LoadException.java (hunk 6 above)
Comment (Joanna-YJA) [reply src/main/java/seedu/address/storage/exceptions/LoadException.java]: What do you mean haha, i couldnt find the extra brace.

File: src/main/java/seedu/address/AlfredException.java (hunk 2 above)
Comment (Joanna-YJA) [reply src/main/java/seedu/address/AlfredException.java]: What do you mean haha, i couldnt find the extra brace.

Comment (hcwong) [reply src/main/java/seedu/address/AlfredException.java]: Ah soz all my comments outdated liao lol, looks good

File: src/main/java/seedu/address/storage/exceptions/LoadException.java (hunk 6 above)
Comment (hcwong) [reply src/main/java/seedu/address/storage/exceptions/LoadException.java]: Same as above, think I commented on the old commit lol 🕐

**Pull request merged**